include/circt/Dialect/SV/Statements.td

```diff
@@ -143,3 +143,43 @@ def FatalOp : SVOp<"fatal"> {
 
   let assemblyFormat = [{ attr-dict }];
 }
+
+//===----------------------------------------------------------------------===//
+// Verification Statements
+//===----------------------------------------------------------------------===//
+
+def AssertOp : SVOp<"assert"> {
+  let summary = "immediate assertion statement";
+  let description = [{
+    Assert that a specific condition is always true.
+  }];
+
+  let arguments = (ins AnyType:$predicate);
+  let results = (outs);
+
+  let assemblyFormat = [{ attr-dict `(` $predicate `)` `:` type($predicate) }];
+}
+
+def AssumeOp : SVOp<"assume"> {
+  let summary = "assume property statement";
+  let description = [{
+    Assume that a specific property is always true.
+  }];
+
+  let arguments = (ins AnyType:$property);
+  let results = (outs);
+
+  let assemblyFormat = [{ attr-dict `(` $property `)` `:` type($property) }];
+}
+
```

Comment:
Since testbenches are going to be made from these, do we want to do something to ensure their [all 3] name?  It looks like tools are going to make reports based on name:
reg_assume : assume property reg_something
since these don't return a value, we might want to embed a name as either a property or argument.

Reply:
Ah, I see, I was confusing property assertions with assertions.  I'm think this is fine as is, but long term, verilog assertions have control flow!  Because of course they do.
```
assert (A == B) 
  $display ("OK.");
else
  $error("It's gone wrong");
```

Reply:
that is wacky :-)

Reply:
Let me hack on this a bit more in this PR. I can add in a format that includes: (1) optional name (stored in an attribute), (2) optional true condition, and (3) optional false condition.

There's also a benefit in lowering from FIRRTL here as we currently don't have any way of printing the string associated with an assertion on failure (which the FIRRTL op includes). This can be packed into the false condition.

We also don't have `$display` or `$error` ops and instead use the legacy FIRRTL lowering of `$fwrite` (with the stderr handle) and `$fatal`. Would the former two be worth packing into this PR?

Reply:
Does SystemVerilog do something with the name?  If not, I'd recommend that we don't proactively add complexity in case it is helpful.

I also think it would be worth handling the basic case (which you have here) separate from the complicated case (which should have two regions like the `if` stmt does).  We can add the complex case when there is a client for it.

Reply:
However, if you want to utilize the complex case from FIRRTL lowering, then *that* would be a client and we should totally do it :-)

Reply:
SystemVerilog (and verification engineers) do use the name in my understanding, so you can have some named assertion:

```
MyCheck: assert(false);
```

The FIRRTL IR isn't capturing this name, nor is there any client that needs it yet. I'll leave this, and the true condition/false condition region stuff, out of this PR and leave it as future work.